### PR TITLE
look-controls enable/disable lifecycle fix

### PR DESCRIPTION
**Pull request by mkeblx**
**Description:**
As look-controls cursor setup only done in init would not work correctly if
changed during runtime.

**Changes proposed:**
-Modify to do on setup + change of enablement state.
-Also, fix for #2635

### CODE REVIEW

File: src/components/look-controls.js [hunk 1]
```diff
@@ -59,6 +52,24 @@ module.exports.Component = registerComponent('look-controls', {
     this.updatePosition();
   },
 
+  setEnabled: function (enabled) {
+    var sceneEl = this.el.sceneEl;
+
+    function enableGrabCursor () {
+      sceneEl.canvas.classList.add('a-grab-cursor');
```
Comment (cvan) [inline src/components/look-controls.js]: IMO this should be a `data-` attribute instead of a class name

File: src/components/look-controls.js [hunk 2]
```diff
@@ -59,6 +52,24 @@ module.exports.Component = registerComponent('look-controls', {
...
+    }
+    function disableGrabCursor () {
+      sceneEl.canvas.classList.remove('a-grab-cursor');
+    }
+
+    if (!sceneEl.canvas) {
+      sceneEl.addEventListener('render-target-loaded',
+        enabled ? enableGrabCursor : disableGrabCursor);
+    } else {
+      enabled ? enableGrabCursor() : disableGrabCursor();
```
Comment (cvan) [inline src/components/look-controls.js]: I'd just use `if` statements

File: src/components/look-controls.js [hunk 3]
```diff
@@ -259,6 +270,8 @@ module.exports.Component = registerComponent('look-controls', {
 
   onMouseDown: function (event) {
     if (!this.data.enabled) { return; }
+    // only handle primary button
```
Comment (cvan) [inline src/components/look-controls.js]: `// Handle only primary button.`

File: src/components/look-controls.js [hunk 4]
```diff
@@ -259,6 +270,8 @@ module.exports.Component = registerComponent('look-controls', {
...
+    if (event.button !== 0) { return; }
```
Comment (cvan) [inline src/components/look-controls.js]: is this universally true for all platforms?

File: src/components/look-controls.js (hunk 2 above)
Comment (dmarcos) [inline src/components/look-controls.js]: yeah let's do an if statement

File: src/components/look-controls.js (hunk 4 above)
Comment (mkeblx) [inline src/components/look-controls.js]: I think so.
> 0: Main button pressed, usually the left button or the un-initialized state

https://developer.mozilla.org/en-US/docs/Web/API/MouseEvent/button

File: src/components/look-controls.js (hunk 2 above)
Comment (mkeblx) [inline src/components/look-controls.js]: Ok

File: src/components/look-controls.js (hunk 1 above)
Comment (mkeblx) [inline src/components/look-controls.js]: I could do that & mod the CSS. Lmk if that is indeed what want

**Comment by dmarcos** [comment]
Thanks!

**Pull request merged**